src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckJavaAgent.kt

```diff
@@ -82,9 +83,9 @@ internal object LincheckJavaAgent {
     private var isBootstrapJarAddedToClasspath = false
 
     /**
-     * TODO
+     * Names of the classes that were instrumented since the last agent installation.
```

Comment:
Please indicate that you use `canonicalClassName` here (the same for similar usages). Please consider reflecting this in the field names.

Reply:
Another option would be to have two value classes for that: `CanonicalClassName` and `InternalClassName`.

Reply:
Or having one `ClassName` value class everywhere with `canonicalClassName` and `internalClassName` properties.

Reply:
Here I added the comment mentioning that it is canonical class name.

> Another option would be to have two value classes for that: CanonicalClassName and InternalClassName.

I really like this idea. But I would prefer to implement it in a separate PR. 
The reason is that there is several places when canonical/internal class names conversion happens, and I need to carefully review them all and enforce unified rules and invariants across all codebase. 

I would like to enforce the invariant that internal class names are only used in the `transformation` module, and they are always converted to canonical class names during byte-code transformation. Using two values classes can help to enforce this rule.

The reasons is that sometimes this conversion can also lead to performance degradation. 
Sometimes, the internal class name is passed into `ManagedStrategy` methods only to be immediately converted into canonical class name. Here is an example:
https://github.com/JetBrains/lincheck/blob/ecede824e9b3399a499c9efb5aba9d69dae91aa9/src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt#L707

Because of this, additional redundant memory allocations happen at runtime (to create new `String`). 

But, as I said, there is several places in the code when this happens. So I think addressing this problem in a separate PR would be easier.

Reply:
Thank you for the detailed explanation; let's raise an issue then.

Reply:
Done: https://github.com/JetBrains/lincheck/issues/328